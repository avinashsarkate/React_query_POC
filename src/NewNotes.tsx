import React, { useState } from "react"
import  NavBar  from "./Navbar/Navbar"
import { NawUserStyled } from "./NewUser.styled"
import { useMutation } from "@tanstack/react-query"
import { UserData } from "./Types"
import { addNotes } from "./API"
import { useNavigate } from "react-router-dom"



const NewNotes: React.FC = () => {
  const [newUserData, setNewUserData] = useState<UserData>({
    title: "",
    body: ""
  })

    const navigate = useNavigate()

  const { mutate } = useMutation({
    mutationFn:(newUserData: UserData) => addNotes(newUserData)
  })

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    const { name, value } = e.target
    setNewUserData((prevData) => ({
      ...prevData,
      [name]: value
    }))
  }

  const handleAddUser = () => {
    mutate(newUserData, {
      onSuccess: () => {
        alert("Data added successfully")
        resetForm()
        navigate("/")
      },
      onError: (error) => {
        alert(`Error adding data: ${error}`)
      }
    })
  }

  const resetForm = () => {
    setNewUserData({
      title: "",
      body: ""
    })
  }

  return (
    <NawUserStyled>
      <div>
        <NavBar />
        <div className="form-container">
          <div className="form">
            <input
              className="input-field"
              value={newUserData.title}
              placeholder="Title"
              onChange={handleInputChange}
              name="title"
            />
            <br />
            <textarea
              className="input-field"
              value={newUserData.body}
              placeholder="Description"
              onChange={handleInputChange}
              name="body"
            />
            <br />
            <button className="submit-button" onClick={handleAddUser}>
              Submit
            </button>
          </div>
        </div>
      </div>
    </NawUserStyled>
  );
};

export default NewNotes;
