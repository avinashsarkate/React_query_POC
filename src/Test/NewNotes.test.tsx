/* eslint-disable testing-library/prefer-screen-queries */
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import NewNotes from "../NewNotes";
import { fireEvent, render } from "@testing-library/react";


jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve({ data: {} })),
 
}))


describe("NewNotes Component", () => {
  const queryClient = new QueryClient();

  const component = (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <NewNotes />
      </BrowserRouter>
    </QueryClientProvider>
  );

  test("Render component without crashing", () => {
    render(component);
  });

  test("to handle the user Input", () => {
    const { getByPlaceholderText } = render(component);

    expect(getByPlaceholderText("Title")).toBeTruthy();
    expect(getByPlaceholderText("Description")).toBeTruthy();
    
  });
  test("Render component", () => {
    const { container } = render(component);
    expect(container).toBeTruthy();
  });

  test("to create snapshot", () => {
    const { asFragment } = render(component);
    expect(asFragment).toMatchSnapshot();
  });
});
