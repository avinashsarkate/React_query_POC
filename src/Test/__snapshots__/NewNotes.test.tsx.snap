// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewNotes Component to create snapshot 1`] = `[Function]`;
